dev/manual_tests/lib/actions.dart

```diff
@@ -0,0 +1,442 @@
+    _focusNode.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FlatButton(
+      focusNode: _focusNode,
+      focusColor: Colors.red,
+      hoverColor: Colors.blue,
+      onPressed: () => _handleOnPressed(),
+      child: Text(widget.name),
+    );
+  }
+}
+
+class FocusDemo extends StatefulWidget {
+  const FocusDemo({Key key}) : super(key: key);
+
+  @override
+  _FocusDemoState createState() => _FocusDemoState();
+}
+
+class _FocusDemoState extends State<FocusDemo> {
+  FocusNode outlineFocus;
+
+  @override
+  void initState() {
+    super.initState();
+    outlineFocus = FocusNode(debugLabel: 'Demo Focus Node');
+  }
+
+  @override
+  void dispose() {
+    outlineFocus.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final TextTheme textTheme = Theme.of(context).textTheme;
+    return InheritedShortcutMap(
+      manager: ShortcutMap(
+        shortcuts: <LogicalKeySet, ActionTag>{
+          LogicalKeySet(LogicalKeyboardKey.tab): const ActionTag('NextFocus'),
+          LogicalKeySet(LogicalKeyboardKey.shiftLeft, LogicalKeyboardKey.tab): const ActionTag('PreviousFocus'),
+          LogicalKeySet(LogicalKeyboardKey.shiftRight, LogicalKeyboardKey.tab): const ActionTag('PreviousFocus'),
+          LogicalKeySet(LogicalKeyboardKey.arrowUp): const DirectionalFocusActionTag(TraversalDirection.up),
+          LogicalKeySet(LogicalKeyboardKey.arrowDown): const DirectionalFocusActionTag(TraversalDirection.down),
+          LogicalKeySet(LogicalKeyboardKey.arrowLeft): const DirectionalFocusActionTag(TraversalDirection.left),
+          LogicalKeySet(LogicalKeyboardKey.arrowRight): const DirectionalFocusActionTag(TraversalDirection.right),
+        },
+      ),
+      child: InheritedActionDispatcher(
+        dispatcher: UndoableActionDispatcher(actions: <String, ActionFactory>{
+          'SetFocus': () => SetFocusAction(),
+          'NextFocus': () => NextFocusAction(),
+          'PreviousFocus': () => PreviousFocusAction(),
+          'DirectionalFocus': () => DirectionalFocusAction(),
+          'Undo': () => kUndoAction,
+          'Redo': () => kRedoAction,
+        }),
+        child: DefaultFocusTraversal(
+          policy: ReadingOrderTraversalPolicy(),
+          child: InheritedShortcutMap(
+            manager: ShortcutMap(
+              shortcuts: <LogicalKeySet, ActionTag>{
+                LogicalKeySet(LogicalKeyboardKey.controlLeft, LogicalKeyboardKey.shiftLeft, LogicalKeyboardKey.keyZ): const ActionTag('Redo'),
+                LogicalKeySet(LogicalKeyboardKey.controlLeft, LogicalKeyboardKey.shiftRight, LogicalKeyboardKey.keyZ): const ActionTag('Redo'),
+                LogicalKeySet(LogicalKeyboardKey.controlRight, LogicalKeyboardKey.shiftLeft, LogicalKeyboardKey.keyZ): const ActionTag('Redo'),
+                LogicalKeySet(LogicalKeyboardKey.controlRight, LogicalKeyboardKey.shiftRight, LogicalKeyboardKey.keyZ): const ActionTag('Redo'),
```

Comment:
ok yeah we definitely need some way to just say `Ctrl-Shift-Z` without having to say every combination of left and right keys.

Reply:
Yes, I agree.  I was thinking of ways to do this, and I think the best way is to just have some special code for some pseudokeys for the keys that commonly appear in more than one place.  This is especially a problem for numpad keys.  At the least, we could handle CTRL, SHIFT, META, and ALT in this way, since it's unlikely that people will want to know which one was pressed.

Reply:
If instead of passing the final map, you passed in a map that is then processed by the widget to generate the underlying data, you could have the LogicalKeySet class take an argument that itself expands to multiple keys, and the LogicalKeySet itself could then expand to the full set of possible key combinations, which you'd then add to the actual map that you use under the hood.

Reply:
Ooh, nice.  I like that.  I'd still have to add a pseudo `LogicalKeyboardKey.shift` that is never actually generated, but can be placed in one of these maps.  That might be odd in some circumstances, though.

Reply:
Done, for Actions and Shortcuts.  Resulted in simplifying `ActionsDispatcher`.